omod-common/src/main/java/org/openmrs/module/webservices/rest/web/OpenmrsClassScanner.java

```diff
@@ -82,30 +82,28 @@ public static OpenmrsClassScanner getInstance() {
 				
 				MetadataReader metadataReader = metadataReaderFactory.getMetadataReader(resource);
 				
-				if (typeFilter.match(metadataReader, metadataReaderFactory)) {
+				if (typeFilter.match(metadataReader, metadataReaderFactory)
+				        && concrete == metadataReader.getClassMetadata().isConcrete()) {
 					
-					if (concrete == metadataReader.getClassMetadata().isConcrete()) {
+					String classname = metadataReader.getClassMetadata().getClassName();
+					
+					try {
 						
-						String classname = metadataReader.getClassMetadata().getClassName();
+						@SuppressWarnings("unchecked")
+						Class<? extends T> metadata = (Class<? extends T>) OpenmrsClassLoader.getInstance().loadClass(
+						    
+						    classname);
```

Comment:
Since you are correcting this code fragment anyway, it is probably also worth removing this empty line in the middle of the brackets.

Reply:
Please could you mention the lines you are referring to, so I can do the modifications.

Reply:
Ok, will do the modification.